src/clients/printer_auto.rb

```diff
@@ -348,25 +347,15 @@ def ReadFileContent(file_name)
 
     def CreateBackupFile(file_name)
       if "" == file_name ||
-          !Printerlib.ExecuteBashCommand(Ops.add("test -f ", file_name))
+          !Printerlib.ExecuteBashCommand("test -f " + file_name.shellescape)
         return true
       end
-      # See "Make a backup" in tools/modify_cupsd_conf how to create a backup file:
-      if Printerlib.ExecuteBashCommand(
-          Ops.add(
-            Ops.add(
-              Ops.add(Ops.add("rpm -V -f ", file_name), " | grep -q '^..5.*"),
-              file_name
-            ),
-            "$'"
-          )
-        )
-        if Printerlib.ExecuteBashCommand(
-            Ops.add(
-              Ops.add(Ops.add(Ops.add("cp -p ", file_name), " "), file_name),
-              ".yast2save"
-            )
-          )
+      # See "Make a backup" in tools/modify_cupsd_conf how to create a backup file.
+      # Intentionally not escaping file_name in the "grep" call:
+      # If there are weird characters in file_name, we might simply make one backup
+      # of it too many which won't hurt.
+      if Printerlib.ExecuteBashCommand("rpm -V -f " + file_name.shellescape + " | grep -q '^..5.*'" + file_name.shellescape + "'$'")
+        if Printerlib.ExecuteBashCommand("cp -p " + file_name.shellescape + " " + file_name.shellescape + ".yast2save")
```

Comment:
well, in this case I prefer just adding it inside string as it is easier to read.
like:
`"cp -p #{file_name.shellescape} #{file_name.shellescape}.yast2save"`

Reply:
also same here with absolute path.

Reply:
The idea was to keep it readable even for the maintainer of this module who isn't quite as fluent in Ruby, so I tried to keep the hard-to-read Ruby parts minimal.